### ISSUE TITLE
[Bug?]: Cannot use vite proxy with `0.4.x`

### ISSUE THREAD

**Issue by srikrsna** [solidjs/solid-start#1180]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

This is my config:

```ts
import { defineConfig } from "@solidjs/start/config";

export default defineConfig({
  server: {
    fs: {
      allow: ["../.."],
    },
    proxy: {
      "/api": {
        target: "http://localhost:3001",
        changeOrigin: true,
      },
    },
  },
  start: {
    ssr: false,
    server: {
      preset: "static",
    },
  },
});
```

The requests that start with `/api` are not forwarded to `3001`.

I also tried moving the config to the new server part:
```ts
import { defineConfig } from "@solidjs/start/config";

export default defineConfig({
  server: {
    fs: {
      allow: ["../.."],
    },
  },
  start: {
    ssr: false,
    server: {
      preset: "static",
      devProxy: {
        "/api": {
          target: "http://localhost:3001",
          changeOrigin: true,
        },
      },
    },
  },
});
```

### Expected behavior 🤔

Requests should be forwarded to configured proxy targets. This use to work prior to `0.4.0` with a similar config:
```ts
import solid from "solid-start/vite";
import { defineConfig } from "vite";

export default defineConfig({
  define: {
    "import.meta.vitest": "undefined",
  },
  plugins: [solid({ ssr: false })],
  server: {
    proxy: {
      "/api": {
        target: "http://localhost:3001",
        changeOrigin: true,
      },
    },
  },
});
```

### Steps to reproduce 🕹

Steps:

1. Use the any proxy route
2. Start the dev server and open the proxy route 
3. Instead of forwarding it will serve the index file.

**Comment by nksaraf** [comment]
Yeah I had removed some of the proxy features initially from vinxi to keep things simple, since a proxy can be just another API route. 

But I might bring it back, most likely as the nitro config and not the vite config.

What is the use case for the proxy? and does it need to be there both in dev and prod?

**Comment by srikrsna** [comment]
I only need it for dev. My use case is to just to forward it to an existing backend. In prod they will be served by the same server,

**Comment by srikrsna** [comment]
> But I might bring it back, most likely as the nitro config and not the vite config.

I think this is a bit confusing, I was initially looking for the `fs` option in the server but had to use the vite one. The proxy config is a similar story

**Comment by vincentfretin** [comment]
I have a similar use case, currently I'm using solid without solid start, but with webpack and webpack dev server with the following configuration for development:
```
  devServer: {
    historyApiFallback: {
      rewrites: [
        {
          from: /^\/scenes\/.*$/,
          to: function (context) {
            return "/public" + context.parsedUrl.pathname;
          },
        }
      ],
    },
    proxy: {
      "/api/tts": {
        target: "http://127.0.0.1:59125",
      },
      "/openai-worker/completions": {
        target: "http://localhost:3000",
        pathRewrite: { "^/openai-worker": "" },
      }
    }
  }
```

"/api/tts" proxy to http://127.0.0.1:59125/api/tts
"/openai-worker/completions" proxy to "http://localhost:3000/completions"
"/scenes/*" proxy to the public/scenes/ folder that is several GB of files.
On production I'm using nginx with similar rules.

I need support for those proxy in development to be able to switch to vite/nitro/vinxi.

**Comment by nksaraf** [comment]
I guess one pending question is, that is it okay for there to be proxies that are dev only (like vite) or proxies should be dev + prod by default, with maybe some way of opting out on either side.. hmm, going to create an issue in vinxi for this.

**Comment by nksaraf** [comment]
Should be fixed with `vinxi:0.0.58`.

You have to use the `start.server.devProxy` option to configure proxies.

```ts

export default defineConfig({
	start: {
		server: {
			devProxy: {
		      "/api/tts": {
		        target: "http://127.0.0.1:59125",
		      },
		      "/openai-worker/completions": {
		        target: "http://localhost:3000/",
		        pathRewrite: { "^/openai-worker": "" },
		      }
			}
		}
	}
})
```
